seismic/inventory/add_time_corrections.py

```diff
@@ -0,0 +1,154 @@
+    print("the first line", all_lines[0])
+    print("the last  line", all_lines[-1])
+
+    line2 = all_lines[1]
+    print("line2=", line2)
+
+    my_items = line2.split(",")
+
+    network_code = my_items[0]  # network_code = "7D"
+    station_code = my_items[1]  # station_code = "DE43"
+
+    return (network_code, station_code, all_lines)
+
+
+def add_gpscorrection_into_stationxml(csv_file, input_xml, out_xml=None):
+    """
+    read in the correction CSV data from a file, get the station metadata node from input_xml file, 
+    then add the CSV data into the station xml node to write into out_xml
+    Args:
+        csv_file: input csv file with correction data
+        input_xml: input original stationXML file which contains the metadata for the network and station of csv_file
+        out_xml:  Directory of the output xml file
+
+    Returns:
+        full path of the output xml file
+
+    """
+
+    ns = "https://github.com/GeoscienceAustralia/hiperseis/xmlns/1.0"
+
+    (net, sta, csv_data) = get_csv_correction_data(csv_file)
+
+    # path2_myxml = "/home/feizhang/Githubz/hiperseis/tests/testdata/7D_2012_2013.xml"
+    my_inv = read_inventory(input_xml)
+
+    # https://docs.obspy.org/packages/autogen/obspy.core.inventory.inventory.Inventory.select.html#obspy.core.inventory.inventory.Inventory.select
+
+    selected_inv = my_inv.select(station=sta)
+
+    print(selected_inv)
+
+    my_tag = AttribDict()
+    my_tag.namespace = ns
+    my_tag.value = csv_data
+
+    selected_inv.networks[0].stations[0].extra = AttribDict()
+    selected_inv.networks[0].stations[0].extra.gpsclockcorrection = my_tag
+
+    stationxml_with_csv = 'modified_inventory_%s.xml' % sta
+
+    if out_xml is not None and os.path.isdir(out_xml):
+        stationxml_with_csv = os.path.join(out_xml, stationxml_with_csv)
+
+    selected_inv.write(stationxml_with_csv, format='STATIONXML',
+                       nsmap={'GeoscienceAustralia': 'https://github.com/GeoscienceAustralia/hiperseis/xmlns/1.0'})
+
+    # my_inv.write('modified_inventory.xml', format='STATIONXML')
+
+    return stationxml_with_csv
+
+
+def extract_csvdata(path2xml):
+    """
+    Read and extract the csv data from an inventory file
+
+    Args:
+        path2xml: path_to_stationxml
+
+    Returns:
+        csv_str
+
+    """
+
+    new_inv = read_inventory(path2xml)
+
+    csv_str = new_inv.networks[0].stations[0].extra.gpsclockcorrection.value
+
+    return csv_str
+
+# ----------------------------------------------------------------------------------------------------------------
+# Quick test code
+# Example How to run:
+# python add_time_corrections.py  /g/data/ha3/Passive/SHARED_DATA/GPS_Clock/corrections/7D.CZ40_clock_correction.csv
+#                                ../../tests/testdata/7D_2012_2013.xml      ~/tmpdir/
+# ----------------------------------------------------------------------------------------------------------------
+if __name__ == "__main__":
```

Comment:
Consider raising a new ticket to add a clean CLI interface here. We typically use the `click` module for this.

Reply:
Please mention ticket number here.

Reply:
Jira Ticket https://gajira.atlassian.net/browse/PV-103